### DISCUSSION TITLE
Solax 5kw Hyrbrid Inverter - Showing 4 Entities

### DISCUSSION THREAD

**Discussion by ezpzfragme** [wills106/homeassistant-solax-modbus#463] [Q&A]
Hello

I am very new to this and have come across a stumbling block, once I install the Solax Inverter Modbus it only shows 4 entities, not the complete list I was expecting.

I believe my settings are all correct and was hoping this was something simple that I overlooked, I can not seem to find a similar issue when searching.

Your help would be appreciated - Bradley

**Comment by Alfred99-HA** [comment]
Actually I had the same problem (and I am also quite new in homeassistant and PV as well). I had found in https://community.home-assistant.io/t/universal-solar-inverter-over-modbus-rs485-tcp-custom-component-growatt-sofar-solax-solis/140143/1063 the same problem and a proposal, actually shut down and power up again the inverter.

From another discussion I read that you might need to enter your SN manually in case it will not be derived in the integration progress.

Shut down and power up worked fine für me - got >220 sensore. Now I need to find a proper card

**Comment by wills106** [comment]
@ezpzfragme Do you get any errors in your error log? Also which Gen Hybrid are you trying to connect to and how?

**Reply by Alfred99-HA** [reply]
Actually my problem ist solved in the meantime - see https://github.com/wills106/homeassistant-solax-modbus/issues/449#issuecomment-1646822588. It seems another gateway listening to the modbus blocked the integration.

The Inverter I am using ist a Solax X3 Hybrid 10kWp, SN starting with H3UE. Soulax cloud shown G1/G2 even though SN idicates G3.

2023-07-17 16:41:03.033 WARNING (MainThread) [custom_components.solax_modbus.plugin_solax] SolaX X3 Hybrid 10.0 kWp: reading serial number from address 0x0 failed; other address may succeed
2023-07-17 16:41:03.057 ERROR (MainThread) [pymodbus.logging] Connection to (192.168.0.228, 502) failed: [Errno 104] Connection reset by peer
2023-07-17 16:41:03.063 WARNING (MainThread) [custom_components.solax_modbus.plugin_solax] SolaX X3 Hybrid 10.0 kWp: attempt to read serialnumber failed at 0x300
Traceback (most recent call last):
File "/config/custom_components/solax_modbus/plugin_solax.py", line 59, in _read_serialnr
inverter_data = hub.read_holding_registers(unit=hub._modbus_addr, address=address, count=7)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Reply by wills106** [reply]
It's been documented for a long time that you can only have one active Mobus connection to the [Inverter](https://github.com/wills106/homeassistant-solax-modbus#known-issues)

> Soulax cloud shown G1/G2 even though SN idicates G3

The X3 Gen3 Hybrid's sometimes show up in the Cloud as a Gen1/Gen2 despite being a Gen3 according to the Modbus docs.
